Source/missiles.cpp

```diff
@@ -1209,7 +1209,7 @@ void AddJester(Missile &missile, const AddMissileParameter &parameter)
 		break;
 	case 6:
 		spell = MIS_TOWN;
-		UseMana(Players[missile._misource], SPL_TOWN);
+		UseMana(*missile.sourcePlayer(), SPL_TOWN);

```

Comment:
Still think `sourcePlayer/Monster` should return references and perform no checks (just an `assert`).
As I understand you plan to do this later, which will be another substantial diff? Why not do this now?

Reply:
I opened a PR for that but I closed it since there wasn't a consensus with @AJenbo 

https://github.com/diasurgical/devilutionX/pull/5054

I am fine with both approaches. I have no strong opinion on this 🙂